2018-06-28 10:28:59.307 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-06-28 10:28:59.392 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-06-28 10:28:59.392 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-06-28 10:28:59.392 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-06-28 10:28:59.392 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-06-29 14:27:55.164 +10:00 [FTL] Application startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: connectionString
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer(DbContextOptionsBuilder optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Identity.API.Startup.<>c__DisplayClass7_0.<ConfigureServices>b__9(DbContextOptionsBuilder operation) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 85
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.DbContextOptionsFactory[TContext](IServiceProvider applicationServiceProvider, Action`2 optionsAction)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProviderEngineScope scope)
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.TestService(IServiceProvider serviceProvider, Type service, ILogger logger, String message, Boolean doThrow) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 126
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.Validate(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 58
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.UseIdentityServer(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 29
   at Identity.API.Startup.Configure(IApplicationBuilder app) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 132
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.Configure(IApplicationBuilder app)
   at Microsoft.AspNetCore.Server.IISIntegration.IISSetupFilter.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.Internal.AutoRequestServicesStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-06-29 14:30:54.787 +10:00 [FTL] Application startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: connectionString
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer(DbContextOptionsBuilder optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Identity.API.Startup.<>c__DisplayClass7_0.<ConfigureServices>b__9(DbContextOptionsBuilder operation) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 85
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.DbContextOptionsFactory[TContext](IServiceProvider applicationServiceProvider, Action`2 optionsAction)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProviderEngineScope scope)
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.TestService(IServiceProvider serviceProvider, Type service, ILogger logger, String message, Boolean doThrow) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 126
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.Validate(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 58
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.UseIdentityServer(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 29
   at Identity.API.Startup.Configure(IApplicationBuilder app) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 132
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.Configure(IApplicationBuilder app)
   at Microsoft.AspNetCore.Server.IISIntegration.IISSetupFilter.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.Internal.AutoRequestServicesStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-06-29 14:30:55.084 +10:00 [ERR] Connection id "0HLETF4D973FA", Request id "0HLETF4D973FA:00000001": An unhandled exception was thrown by the application.
System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IO.StreamWriter.Write(String value)
   at Microsoft.AspNetCore.Hosting.Views.ErrorPage.ExecuteAsync()
   at Microsoft.Extensions.RazorViews.BaseView.ExecuteAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2018-06-29 14:30:55.240 +10:00 [ERR] Connection id "0HLETF4D973FB", Request id "0HLETF4D973FB:00000001": An unhandled exception was thrown by the application.
System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IO.StreamWriter.Write(String value)
   at Microsoft.AspNetCore.Hosting.Views.ErrorPage.ExecuteAsync()
   at Microsoft.Extensions.RazorViews.BaseView.ExecuteAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2018-06-29 14:30:55.240 +10:00 [ERR] Connection id "0HLETF4D973FC", Request id "0HLETF4D973FC:00000001": An unhandled exception was thrown by the application.
System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IO.StreamWriter.Write(String value)
   at Microsoft.AspNetCore.Hosting.Views.ErrorPage.ExecuteAsync()
   at Microsoft.Extensions.RazorViews.BaseView.ExecuteAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2018-06-29 14:32:45.949 +10:00 [FTL] Application startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: connectionString
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer(DbContextOptionsBuilder optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Identity.API.Startup.<>c__DisplayClass7_0.<ConfigureServices>b__9(DbContextOptionsBuilder operation) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 85
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.DbContextOptionsFactory[TContext](IServiceProvider applicationServiceProvider, Action`2 optionsAction)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProviderEngineScope scope)
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.TestService(IServiceProvider serviceProvider, Type service, ILogger logger, String message, Boolean doThrow) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 126
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.Validate(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 58
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.UseIdentityServer(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 29
   at Identity.API.Startup.Configure(IApplicationBuilder app) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 132
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.Configure(IApplicationBuilder app)
   at Microsoft.AspNetCore.Server.IISIntegration.IISSetupFilter.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.Internal.AutoRequestServicesStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-06-29 14:50:02.946 +10:00 [FTL] Application startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: connectionString
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer(DbContextOptionsBuilder optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Identity.API.Startup.<>c__DisplayClass7_0.<ConfigureServices>b__9(DbContextOptionsBuilder operation) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 85
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.DbContextOptionsFactory[TContext](IServiceProvider applicationServiceProvider, Action`2 optionsAction)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProviderEngineScope scope)
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.TestService(IServiceProvider serviceProvider, Type service, ILogger logger, String message, Boolean doThrow) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 126
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.Validate(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 58
   at Microsoft.AspNetCore.Builder.IdentityServerApplicationBuilderExtensions.UseIdentityServer(IApplicationBuilder app) in C:\local\identity\server4\IdentityServer4\src\IdentityServer4\Configuration\IdentityServerApplicationBuilderExtensions.cs:line 29
   at Identity.API.Startup.Configure(IApplicationBuilder app) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 132
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.Configure(IApplicationBuilder app)
   at Microsoft.AspNetCore.Server.IISIntegration.IISSetupFilter.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.Internal.AutoRequestServicesStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-06-29 14:52:46.155 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-06-29 14:52:46.201 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-06-29 14:52:46.201 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-06-29 14:52:46.201 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-06-29 14:52:46.201 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-06-29 15:32:34.343 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-06-29 15:32:34.381 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-06-29 15:32:34.381 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-06-29 15:32:34.381 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-06-29 15:32:34.381 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-06-29 15:32:42.308 +10:00 [INF] AuthenticationScheme: Identity.Application was challenged.
2018-06-29 15:33:36.707 +10:00 [ERR] An error occurred using the connection to database 'LibraryBuddy.IdentityDb' on server 'tcp:librarybuddyapp.database.windows.net,1433'.
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) ---> System.ComponentModel.Win32Exception (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnectionAsync(Boolean errorsExpected, CancellationToken cancellationToken)
ClientConnectionId:00000000-0000-0000-0000-000000000000
Error Number:10060,State:0,Class:20
2018-06-29 15:33:37.348 +10:00 [ERR] An error occurred using the connection to database 'LibraryBuddy.IdentityDb' on server 'tcp:librarybuddyapp.database.windows.net,1433'.
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) ---> System.ComponentModel.Win32Exception (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.OpenAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnectionAsync(Boolean errorsExpected, CancellationToken cancellationToken)
ClientConnectionId:00000000-0000-0000-0000-000000000000
Error Number:10060,State:0,Class:20
2018-06-29 15:33:38.437 +10:00 [ERR] An error occurred using the connection to database 'LibraryBuddy.IdentityDb' on server 'tcp:librarybuddyapp.database.windows.net,1433'.
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) ---> System.ComponentModel.Win32Exception (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.OpenAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnectionAsync(Boolean errorsExpected, CancellationToken cancellationToken)
ClientConnectionId:00000000-0000-0000-0000-000000000000
Error Number:10060,State:0,Class:20
2018-06-29 15:33:41.712 +10:00 [ERR] An error occurred using the connection to database 'LibraryBuddy.IdentityDb' on server 'tcp:librarybuddyapp.database.windows.net,1433'.
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) ---> System.ComponentModel.Win32Exception (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.OpenAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnectionAsync(Boolean errorsExpected, CancellationToken cancellationToken)
ClientConnectionId:00000000-0000-0000-0000-000000000000
Error Number:10060,State:0,Class:20
2018-06-29 15:34:31.151 +10:00 [ERR] An error occurred using the connection to database 'LibraryBuddy.IdentityDb' on server 'tcp:librarybuddyapp.database.windows.net,1433'.
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.) ---> System.ComponentModel.Win32Exception (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
   at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnectionAsync(Boolean errorsExpected, CancellationToken cancellationToken)
ClientConnectionId:00000000-0000-0000-0000-000000000000
Error Number:10060,State:0,Class:20
2018-07-02 01:05:38.359 +10:00 [FTL] Application startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: configuration
   at StackExchange.Redis.ConnectionMultiplexer.CreateMultiplexer(Object configuration) in c:\code\StackExchange.Redis\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 844
   at StackExchange.Redis.ConnectionMultiplexer.<>c__DisplayClass75_0.<Connect>b__0() in c:\code\StackExchange.Redis\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 855
   at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl(Func`1 multiplexerFactory, TextWriter log) in c:\code\StackExchange.Redis\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 871
   at StackExchange.Redis.ConnectionMultiplexer.Connect(String configuration, TextWriter log) in c:\code\StackExchange.Redis\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 855
   at Identity.API.Startup.ConfigureServices(IServiceCollection services) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 111
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-07-02 01:05:39.716 +10:00 [ERR] Connection id "0HLEVCGDLN47A", Request id "0HLEVCGDLN47A:00000001": An unhandled exception was thrown by the application.
System.ObjectDisposedException: Cannot write to the response body, the response has completed.
Object name: 'HttpResponseStream'.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.ValidateState(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Write(String value)
   at Microsoft.AspNetCore.Hosting.Views.ErrorPage.ExecuteAsync()
   at Microsoft.Extensions.RazorViews.BaseView.ExecuteAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2018-07-02 01:07:23.298 +10:00 [WRN] No XML encryptor configured. Key "01fb7c87-8d6c-4eee-a896-f5d5573985d7" may be persisted to storage in unencrypted form.
2018-07-02 01:07:24.791 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-07-02 01:07:24.791 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-07-02 01:07:24.791 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-07-02 01:07:24.791 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-07-02 01:07:24.792 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-07-03 12:28:55.240 +10:00 [WRN] No XML encryptor configured. Key "b10db427-47a1-430e-9c2f-9acc59aa6e80" may be persisted to storage in unencrypted form.
2018-07-03 12:28:56.086 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-07-03 12:28:56.086 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-07-03 12:28:56.086 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-07-03 12:28:56.086 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-07-03 12:28:56.086 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-07-03 12:33:00.528 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-07-03 12:33:00.568 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-07-03 12:33:00.568 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-07-03 12:33:00.568 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-07-03 12:33:00.568 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-07-15 23:01:40.027 +10:00 [INF] Migrating database associated with the context PersistedGrantDbContext
2018-07-15 23:01:56.346 +10:00 [INF] An error occurred while migrating database on context PersistedGrantDbContext
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.<>c__DisplayClass0_1`1.<MigrateDbContext>b__0() in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 35
   at Polly.Policy.<>c__DisplayClass144_0.<Execute>b__0(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 20
   at Polly.RetrySyntax.<>c__DisplayClass20_1.<WaitAndRetry>b__1(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 451
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) in C:\projects\polly\src\Polly.Shared\Retry\RetryEngine.cs:line 31
   at Polly.RetrySyntax.<>c__DisplayClass20_0.<WaitAndRetry>b__0(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 450
   at Polly.Policy.ExecuteInternal(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.cs:line 43
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 83
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.MigrateDbContext[TContext](IWebHost webHost, Action`2 seed) in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 33
2018-07-15 23:01:58.181 +10:00 [INF] Migrating database associated with the context ApplicationDbContext
2018-07-15 23:02:48.549 +10:00 [INF] Migrating database associated with the context PersistedGrantDbContext
2018-07-15 23:02:48.584 +10:00 [INF] An error occurred while migrating database on context PersistedGrantDbContext
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.<>c__DisplayClass0_1`1.<MigrateDbContext>b__0() in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 35
   at Polly.Policy.<>c__DisplayClass144_0.<Execute>b__0(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 20
   at Polly.RetrySyntax.<>c__DisplayClass20_1.<WaitAndRetry>b__1(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 451
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) in C:\projects\polly\src\Polly.Shared\Retry\RetryEngine.cs:line 31
   at Polly.RetrySyntax.<>c__DisplayClass20_0.<WaitAndRetry>b__0(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 450
   at Polly.Policy.ExecuteInternal(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.cs:line 43
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 83
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.MigrateDbContext[TContext](IWebHost webHost, Action`2 seed) in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 33
2018-07-15 23:02:48.839 +10:00 [INF] Migrating database associated with the context ApplicationDbContext
2018-07-15 23:02:49.856 +10:00 [INF] An error occurred while migrating database on context ApplicationDbContext
System.InvalidOperationException: Navigation property 'StreetAdress' on entity type 'ApplicationUser' is not virtual. UseLazyLoadingProxies requires all entity types to be public, unsealed, have virtual navigation properties, and have a public or protected constructor.
   at Microsoft.EntityFrameworkCore.Proxies.Internal.ProxyBindingRewriter.Apply(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnModelBuilt(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, IModelValidator validator)
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.GetRelationalService[TService](IInfrastructure`1 databaseFacade)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.Migrate(DatabaseFacade databaseFacade)
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.<>c__DisplayClass0_1`1.<MigrateDbContext>b__0() in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 35
   at Polly.Policy.<>c__DisplayClass144_0.<Execute>b__0(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 20
   at Polly.RetrySyntax.<>c__DisplayClass20_1.<WaitAndRetry>b__1(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 451
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) in C:\projects\polly\src\Polly.Shared\Retry\RetryEngine.cs:line 31
   at Polly.RetrySyntax.<>c__DisplayClass20_0.<WaitAndRetry>b__0(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 450
   at Polly.Policy.ExecuteInternal(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.cs:line 43
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 83
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.MigrateDbContext[TContext](IWebHost webHost, Action`2 seed) in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 33
2018-07-15 23:02:49.886 +10:00 [INF] Migrating database associated with the context ConfigurationDbContext
2018-07-15 23:02:49.887 +10:00 [INF] An error occurred while migrating database on context ConfigurationDbContext
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.<>c__DisplayClass0_1`1.<MigrateDbContext>b__0() in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 35
   at Polly.Policy.<>c__DisplayClass144_0.<Execute>b__0(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 20
   at Polly.RetrySyntax.<>c__DisplayClass20_1.<WaitAndRetry>b__1(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 451
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) in C:\projects\polly\src\Polly.Shared\Retry\RetryEngine.cs:line 31
   at Polly.RetrySyntax.<>c__DisplayClass20_0.<WaitAndRetry>b__0(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 450
   at Polly.Policy.ExecuteInternal(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.cs:line 43
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 83
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.MigrateDbContext[TContext](IWebHost webHost, Action`2 seed) in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 33
2018-07-15 23:02:49.963 +10:00 [FTL] Application startup exception
System.NotImplementedException: The method or operation is not implemented.
   at LibraryBuddy.Services.Identity.API.Infrastructure.CommonStartup.ConfigureServices(IServiceCollection services, IConfiguration configuration) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Infrastructure\CommonStartup.cs:line 20
   at LibraryBuddy.Services.Identity.API.Infrastructure.Extensions.ServiceCollectionExtensions.ConfigureServices(IServiceCollection services, IConfiguration configuration) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Infrastructure\Extensions\ServiceCollectionExtensions.cs:line 46
   at LibraryBuddy.Services.Identity.API.Infrastructure.Extensions.ServiceCollectionExtensions.ConfigureApplicationServices(IServiceCollection services, IConfiguration configuration) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Infrastructure\Extensions\ServiceCollectionExtensions.cs:line 31
   at LibraryBuddy.Services.Identity.API.Startup.ConfigureServices(IServiceCollection services) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Startup.cs:line 39
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
2018-07-15 23:56:04.005 +10:00 [INF] Migrating database associated with the context PersistedGrantDbContext
2018-07-15 23:56:37.315 +10:00 [INF] Migrating database associated with the context ApplicationDbContext
2018-07-15 23:57:30.797 +10:00 [INF] Migrating database associated with the context ConfigurationDbContext
2018-07-15 23:58:36.343 +10:00 [INF] An error occurred while migrating database on context ConfigurationDbContext
System.AggregateException: One or more errors occurred. (Value cannot be null.
Parameter name: key) ---> System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.TryGetEntry(Object entity, Boolean throwOnNonUniqueness)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NavigationFixer.InitialFixup(InternalEntityEntry entry, ISet`1 handledForeignKeys, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NavigationFixer.StateChanged(InternalEntityEntry entry, EntityState oldState, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntryNotifier.StateChanged(InternalEntityEntry entry, EntityState oldState, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.FireStateChanged(EntityState oldState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode node, Boolean force)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode node, TState state, Func`3 handleNode)
   at Microsoft.EntityFrameworkCore.DbContext.SetEntityState[TEntity](TEntity entity, EntityState entityState)
   at LibraryBuddy.Services.Identity.API.Data.ConfigurationDbContextSeed.SeedAsync(ConfigurationDbContext context, IConfiguration configuration) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Data\ConfigurationDbContextSeed.cs:line 37
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at LibraryBuddy.Services.Identity.API.Program.<>c.<Main>b__0_2(ConfigurationDbContext context, IServiceProvider services) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Program.cs:line 56
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.<>c__DisplayClass0_1`1.<MigrateDbContext>b__0() in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 37
   at Polly.Policy.<>c__DisplayClass144_0.<Execute>b__0(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 20
   at Polly.RetrySyntax.<>c__DisplayClass20_1.<WaitAndRetry>b__1(Context ctx, CancellationToken ct) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 451
   at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, IEnumerable`1 shouldRetryExceptionPredicates, IEnumerable`1 shouldRetryResultPredicates, Func`1 policyStateFactory) in C:\projects\polly\src\Polly.Shared\Retry\RetryEngine.cs:line 31
   at Polly.RetrySyntax.<>c__DisplayClass20_0.<WaitAndRetry>b__0(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Retry\RetrySyntax.cs:line 450
   at Polly.Policy.ExecuteInternal(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.cs:line 43
   at Polly.Policy.Execute(Action`2 action, Context context, CancellationToken cancellationToken) in C:\projects\polly\src\Polly.Shared\Policy.ExecuteOverloads.cs:line 83
   at Microsoft.AspNetCore.Hosting.IWebHostExtension.MigrateDbContext[TContext](IWebHost webHost, Action`2 seed) in D:\projects\LibraryBuddy\LibraryBuddy\src\BuildingBlocks\WebHost\WebHost.Customization\WebHostExtension.cs:line 33
---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.TryGetEntry(Object entity, Boolean throwOnNonUniqueness)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NavigationFixer.InitialFixup(InternalEntityEntry entry, ISet`1 handledForeignKeys, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NavigationFixer.StateChanged(InternalEntityEntry entry, EntityState oldState, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntryNotifier.StateChanged(InternalEntityEntry entry, EntityState oldState, Boolean fromQuery)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.FireStateChanged(EntityState oldState)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode node, Boolean force)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode node, TState state, Func`3 handleNode)
   at Microsoft.EntityFrameworkCore.DbContext.SetEntityState[TEntity](TEntity entity, EntityState entityState)
   at LibraryBuddy.Services.Identity.API.Data.ConfigurationDbContextSeed.SeedAsync(ConfigurationDbContext context, IConfiguration configuration) in D:\projects\LibraryBuddy\LibraryBuddy\src\Services\Identity\Identity.API\Data\ConfigurationDbContextSeed.cs:line 37<---

2018-07-15 23:58:36.644 +10:00 [WRN] No XML encryptor configured. Key "cd83e753-7051-4c86-9ef1-329409cf4eda" may be persisted to storage in unencrypted form.
2018-07-15 23:58:52.329 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-07-15 23:58:52.330 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-07-15 23:58:52.330 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-07-15 23:58:52.330 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-07-15 23:58:52.331 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-07-15 23:58:52.808 +10:00 [DBG] Starting token cleanup
2018-07-16 00:07:07.014 +10:00 [INF] Migrating database associated with the context PersistedGrantDbContext
2018-07-16 00:07:11.710 +10:00 [INF] Migrating database associated with the context ApplicationDbContext
2018-07-16 00:07:16.105 +10:00 [INF] Migrating database associated with the context ConfigurationDbContext
2018-07-16 00:07:22.257 +10:00 [DBG] Using Identity.Application as default scheme for authentication
2018-07-16 00:07:22.262 +10:00 [DBG] Using Identity.External as default scheme for sign-in
2018-07-16 00:07:22.263 +10:00 [DBG] Using Identity.External as default scheme for sign-out
2018-07-16 00:07:22.264 +10:00 [DBG] Using Identity.Application as default scheme for challenge
2018-07-16 00:07:22.264 +10:00 [DBG] Using Identity.Application as default scheme for forbid
2018-07-16 00:07:22.663 +10:00 [DBG] Starting token cleanup
